### PR TITLE
[Sql] Overhaul failover group cmdlets to use current API and support multiple secondaries

### PR DESCRIPTION
**Pull request by alecbain**
## Description

This changeset updates supporting failover group commands (Add/Remove database from failover group and Get) to function with the new mutli-secondary feature by using the current client instead of the legacy client. 

In addition, updated Get command to display multiple partner servers properties when relevant. Get command will now display PartnerServers list when the failover group contains multiple secondaries, otherwise it will display using the old PartnerServer and PartnerResourceGroup as it previously functioned.

## Mandatory Checklist

- Please choose the target release of Azure PowerShell
  - [x] [General release](../blob/main/CONTRIBUTING.md#target-release-types)
  - [ ] [Public preview](../blob/main/CONTRIBUTING.md#target-release-types)
  - [ ] [Private preview](../blob/main/CONTRIBUTING.md#target-release-types)
  - [ ] [Engineering build](../blob/main/CONTRIBUTING.md#target-release-types)
  - [ ] N/A

- [x] Check this box to confirm: **I have read the [_Submitting Changes_](../blob/main/CONTRIBUTING.md#submitting-changes) section of [`CONTRIBUTING.md`](../blob/main/CONTRIBUTING.md) and reviewed the following information:**

* **SHOULD** update `ChangeLog.md` file(s) appropriately
    * For SDK-based development mode, update `src/{{SERVICE}}/{{SERVICE}}/ChangeLog.md`.
        * A snippet outlining the change(s) made in the PR should be written under the `## Upcoming Release` header in the past tense. 
    * For autorest-based development mode, include the changelog in the PR description.
    * Should **not** change `ChangeLog.md` if no new release is required, such as fixing test case only.
* **SHOULD** regenerate markdown help files if there is cmdlet API change. [Instruction](../blob/main/documentation/development-docs/help-generation.md#updating-all-markdown-files-in-a-module)
* **SHOULD** have proper test coverage for changes in pull request.
* **SHOULD NOT** adjust version of module manually in pull request

### CODE REVIEW

File: src/Sql/Sql/ChangeLog.md [hunk 1]
```diff
@@ -19,6 +19,8 @@
 -->
 ## Upcoming Release
 * Added `DatabaseFormat` and `PricingModel` parameters to `New-AzSqlInstance`, `Set-AzSqlInstance`
+* Added multi-secondary support for `Get-AzSqlDatabaseFailoverGroup`, `Remove-AzSqlDatabaseFromFailoverGroup` and `Add-AzSqlDatabaseFromFailoverGroup`
+* Change default FailoverPolicy value for `New-AzSqlDatabaseFailoverGroup`, `Set-AzSqlDatabaseFailoverGroup`
```
Comment (wyunchi-ms) [inline src/Sql/Sql/ChangeLog.md:23]:
```suggestion
* Changed default FailoverPolicy value for `New-AzSqlDatabaseFailoverGroup`, `Set-AzSqlDatabaseFailoverGroup` from `Automatic` to `Manual`
```

**Comment by wyunchi-ms** [comment]
Changing default value will cause breaking change for customers. We need to do it in major release.

File: src/Sql/Sql/Failover Group/Cmdlet/NewAzureSqlFailoverGroup.cs [hunk 2]
```diff
@@ -83,7 +83,7 @@ public class NewAzureSqlFailoverGroup : AzureSqlFailoverGroupCmdletBase
         [Parameter(Mandatory = false,
             HelpMessage = "The failover policy of the Azure SQL Database Failover Group.")]
         [ValidateNotNullOrEmpty]
-        [PSDefaultValue(Help = "Automatic")]
+        [PSDefaultValue(Help = "Manual")]
```
Comment (wyunchi-ms) [inline src/Sql/Sql/Failover Group/Cmdlet/NewAzureSqlFailoverGroup.cs:86]:
```suggestion
        [PSDefaultValue(Help = "Manual", Value = FailoverPolicy.Manual)]
```

File: src/Sql/Sql/Failover Group/Cmdlet/SetAzureSqlFailoverGroup.cs [hunk 3]
```diff
@@ -59,7 +59,7 @@ public class SetAzureSqlFailoverGroup : AzureSqlFailoverGroupCmdletBase
         [Parameter(Mandatory = false,
             HelpMessage = "The failover policy of the Azure SQL Database Failover Group.")]
         [ValidateNotNullOrEmpty]
-        [PSDefaultValue(Help = "Automatic")]
+        [PSDefaultValue(Help = "Manual")]
```
Comment (wyunchi-ms) [inline src/Sql/Sql/Failover Group/Cmdlet/SetAzureSqlFailoverGroup.cs:62]:
```suggestion
        [PSDefaultValue(Help = "Manual", Value = FailoverPolicy.Manual)]
```

**Comment by wyunchi-ms** [comment]
Sumbit a PR to add the warning messages: https://github.com/Azure/azure-powershell/pull/23923.
This PR will be merged in major release, and please update the help file.

**Comment by YanaXu** [comment]
@alecbain , could you please resolve the conflicts?

File: src/Sql/Sql/ChangeLog.md [hunk 4]
```diff
@@ -28,6 +28,8 @@
 * Added a new cmdlet for Azure SQL Managed Instance refresh external governance status
   - 'Invoke-AzSqlInstanceExternalGovernanceStatusRefresh'
 * Updated `Get-AzSqlInstance` to support returning the `ExternalGovernanceStatus` property
+* Added multi-secondary support for `Get-AzSqlDatabaseFailoverGroup`, `Remove-AzSqlDatabaseFromFailoverGroup` and `Add-AzSqlDatabaseFromFailoverGroup`
```
Comment (YanaXu) [inline src/Sql/Sql/ChangeLog.md:31]: Move it to "## Upcoming Release"

Comment (alecbain) [reply src/Sql/Sql/ChangeLog.md]: Good catch, bumped it back up to upcoming release in latest revision.

File: src/Sql/Sql/Failover Group/Cmdlet/NewAzureSqlFailoverGroup.cs [hunk 5]
```diff
@@ -85,7 +85,7 @@ public class NewAzureSqlFailoverGroup : AzureSqlFailoverGroupCmdletBase
         [Parameter(Mandatory = false,
             HelpMessage = "The failover policy of the Azure SQL Database Failover Group.")]
         [ValidateNotNullOrEmpty]
-        [PSDefaultValue(Help = "Automatic")]
+        [PSDefaultValue(Help = "Manual", Value = FailoverPolicy.Manual)]
```
Comment (YanaXu) [inline src/Sql/Sql/Failover Group/Cmdlet/NewAzureSqlFailoverGroup.cs:88]: Please update the help md file accordingly. Please update it manually.

File: src/Sql/Sql/Failover Group/Cmdlet/SetAzureSqlFailoverGroup.cs [hunk 6]
```diff
@@ -61,7 +61,7 @@ public class SetAzureSqlFailoverGroup : AzureSqlFailoverGroupCmdletBase
         [Parameter(Mandatory = false,
             HelpMessage = "The failover policy of the Azure SQL Database Failover Group.")]
         [ValidateNotNullOrEmpty]
-        [PSDefaultValue(Help = "Automatic")]
+        [PSDefaultValue(Help = "Manual", Value = FailoverPolicy.Manual)]
```
Comment (YanaXu) [inline src/Sql/Sql/Failover Group/Cmdlet/SetAzureSqlFailoverGroup.cs:64]: Please update the help md file accordingly. Please update it manually.

**Review by YanaXu — APPROVED** [review]

**Pull request merged**